pandas/core/arrays/integer.py

```diff
@@ -219,25 +220,33 @@ def __init__(self, values, mask=None, dtype=None, copy=False):
         -------
         IntegerArray
         """
-        self._data, self._mask = coerce_to_array(
-            values, dtype=dtype, mask=mask, copy=copy)
+        if not (isinstance(values, np.ndarray)
+                and np.issubdtype(values.dtype, np.integer)):
+            raise TypeError("values should be integer numpy array")
+        if not (isinstance(mask, np.ndarray) and mask.dtype == np.bool_):
+            raise TypeError("mask should be boolean numpy array")
+
+        if copy:
+            values = values.copy()
+            mask = mask.copy()
+
+        self._data = values
+        self._mask = mask
 
     @classmethod
     def _from_sequence(cls, scalars, dtype=None, copy=False):
-        return cls(scalars, dtype=dtype, copy=copy)
+        return integer_array(scalars, dtype=dtype, copy=copy)
 
     @classmethod
     def _from_factorized(cls, values, original):
-        return cls(values, dtype=original.dtype)
+        return integer_array(values, dtype=original.dtype)
 
     def __getitem__(self, item):
         if is_integer(item):
             if self._mask[item]:
                 return self.dtype.na_value
             return self._data[item]
-        return type(self)(self._data[item],
-                          mask=self._mask[item],
-                          dtype=self.dtype)
+        return type(self)(self._data[item], self._mask[item])
```

Comment:
@jreback additional question: what do you find of writing `IntegerArray(...)` instead of `type(self)(...)` ? 

Python perfectly allows that (and is the same here, as we don't subclass this one further), and I personally find that easier to read.

Reply:
I like using ``type()`` as I expect this to be subclasses for ``BooleanArray``, though it could be that this not needed for that, but want to keep open that possiblitiy.